lightglue/disk.py

```diff
@@ -45,3 +55,15 @@ def forward(self, data: dict) -> dict:
             'keypoint_scores': scores.to(image),
             'descriptors': descriptors.to(image),
         }
+
+    def extract(self, img: torch.Tensor, **conf) -> dict:
+        """ Perform extraction with online resizing"""
+        if img.dim() == 3:
+            img = img[None]  # add batch dim
+        assert img.dim() == 4 and img.shape[0] == 1
+        shape = img.shape[-1], img.shape[-2]
+        img, scales = ImagePreprocessor(**self.preprocess_conf, **conf)(img)
```

Comment:
Merge the dictionaries before expanding such that conf can override resize (and potentially disable it with None)